### PR TITLE
feat(PoolPage): When set input value, calculate the other input value. Validate if pool ratio is right

### LINKED ISSUE #148: Validate pool creation values
When creating a pool with existing liquidity the user needs to provide an exact ratio of the current on reserves;

We can validate on the front-end by using the pool info;
```
(eth_reserve / token_reserve) === (eth_inputed by the user / token_inputed by the user)
```
**Note**; This should also consider slippage when its implemented: https://github.com/FuelLabs/swayswap/issues/147

### LINKED ISSUE #179: Auto calculate inputs on add liquidity
> **Note**
>  This only applies when the pool has liquidity.
> (eth_reserver > 0 && token_reserve > 0)

When adding liquidity to a pool with liquidity, the user can't set the ratio freely; because of that, we should auto calculate the value of one input based on the value changed on the other. 

Meta code;
```ts
const ratio_eth_token = (eth_reserver/token_reserve);
const ratio_token_eth = (token_reserve/eth_reserver);
const token_amount = eth_user_input * ratio_eth_token;
const eth_amount = token_user_input * ratio_token_eth;
```

This is a similar behavior from the swap page, but we do not need to invert selection in this case.

### PR DESCRIPTION
**Pull request by LuizAsFight**
- When set one token value, calculate the other one according to `reserve ratio` we have
- Validate `add_liquidity` ratio is right

Closes #179
Closes #148

### CODE REVIEW

File: packages/app/src/config.ts [hunk 1]
```diff
@@ -11,9 +11,10 @@ export const TOKEN_ID = import.meta.env.VITE_TOKEN_ID!;
 export const DECIMAL_UNITS = 3;
 export const FAUCET_AMOUNT = parseUnits('0.5', DECIMAL_UNITS).toBigInt();
 export const MINT_AMOUNT = parseUnits('2000', DECIMAL_UNITS).toBigInt();
-export const ONE_ASSET = parseUnits('1', DECIMAL_UNITS).toBigInt();
+export const ONE_ASSET_UNIT = parseUnits('1', DECIMAL_UNITS).toBigInt();
```
Comment (luizstacio) [inline packages/app/src/config.ts:14]: This is not one unit, is ONE asset, a unit is a `1` one asset is `1000`

File: packages/app/src/lib/asset.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { toNumber } from "fuels"
+import { ONE_ASSET_UNIT } from "~/config"
+
+export const calculateRatio = (fromAmount?: bigint | null, toAmount?: bigint | null) => {
+  const _fromAmount = fromAmount || BigInt(0);
```
Comment (luizstacio) [inline packages/app/src/lib/asset.ts:5]: Maybe is safer to use 1 as default instead of 0?

File: packages/app/src/pages/PoolPage/AddLiquidity.tsx [hunk 3]
```diff
@@ -143,26 +163,34 @@ export default function AddLiquidity() {
...
-  const handleCreatePool = () => {
-    const fromAmount = fromInput.amount;
-    const toAmount = toInput.amount;
+  const validateCreatePool = () => {
+    const errors = [];
 
-    if (!fromAmount) {
-      throw new Error('"fromAmount" is required');
+    if (!fromInput.amount) {
+      errors.push(`Inform ${coinFrom.name} amount`);
```
Comment (luizstacio) [inline packages/app/src/pages/PoolPage/AddLiquidity.tsx:170]: Enter ${x} amount

File: packages/app/src/pages/PoolPage/AddLiquidity.tsx [hunk 4]
```diff
@@ -143,26 +163,34 @@ export default function AddLiquidity() {
...
     }
     if (!toInput.hasEnoughBalance) {
-      throw new Error(`Insufficient ${coinTo.name} balance`);
+      errors.push(`Insufficient ${coinTo.name} balance`);
     }
+    if (poolInfo) {
+      const minRatio = reservesFromToRatio - (reservesFromToRatio * SLIPPAGE_TOLERANCE);
+      const maxRatio = reservesFromToRatio + (reservesFromToRatio * SLIPPAGE_TOLERANCE);
+
+      if ( addLiquidityRatio < minRatio || addLiquidityRatio > maxRatio ) {
```
Comment (luizstacio) [inline packages/app/src/pages/PoolPage/AddLiquidity.tsx:185]: I think is show be something more like

```
const minRatio = reservesFromToRatio * (1 - SLIPPAGE_TOLERANCE);
const maxRatio = reservesFromToRatio * (1 - SLIPPAGE_TOLERANCE);
```

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: packages/app/src/lib/asset.ts (hunk 2 above)
Comment (LuizAsFight) [reply packages/app/src/lib/asset.ts:5]: I'll test

Comment (luizstacio) [reply packages/app/src/lib/asset.ts:5]: Need to change both

**Review by luizstacio — CHANGES_REQUESTED** [review]

**Comment by LuizAsFight** [comment]
@luizstacio I changed both to zero again to make function accept any values (undefined / null / zero)... add validation to make sure it's not NaN of Infinity

**Review by luizstacio — APPROVED** [review]

**Pull request merged**